cmd/algons/dnsaddrCmd.go

```diff
@@ -63,3 +80,93 @@ var dnsaddrTreeCmd = &cobra.Command{
 		}
 	},
 }
+var dnsaddrTreeDeleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Recursively resolves and deletes the dnsaddr entries of the given domain",
+	Long:  "Recursively resolves and deletes the dnsaddr entries of the given domain",
+	Run: func(cmd *cobra.Command, args []string) {
+		addr, err := multiaddr.NewMultiaddr(fmt.Sprintf("/dnsaddr/%s", dnsaddrDomain))
+		if err != nil {
+			fmt.Printf("unable to construct multiaddr for %s : %v\n", dnsaddrDomain, err)
+			return
+		}
+		controller := dnsaddr.NewMultiaddrDNSResolveController(secure, "")
+		cfZoneID, cfToken, err := getClouldflareCredentials()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error getting DNS credentials: %v", err)
+			return
+		}
+		cloudflareDNS := cloudflare.NewDNS(cfZoneID, cfToken)
+		var recordsToDelete []cloudflare.DNSRecordResponseEntry
+		err = dnsaddr.Iterate(addr, controller, func(entryFrom multiaddr.Multiaddr, entries []multiaddr.Multiaddr) error {
+			domain, _ := entryFrom.ValueForProtocol(multiaddr.P_DNSADDR)
+			name := fmt.Sprintf("_dnsaddr.%s", domain)
+			fmt.Printf("listing records for %s\n", name)
+			records, err0 := cloudflareDNS.ListDNSRecord(context.Background(), "TXT", name, "", "", "", "")
+			if err0 != nil {
+				fmt.Printf("erroring listing dns records for %s %s\n", domain, err)
+				return err
+			}
+			for _, record := range records {
+				fmt.Printf("found record to delete %s:%s\n", record.Name, record.Content)
+				recordsToDelete = append(recordsToDelete, record)
+			}
+			return nil
+		})
+		if err != nil {
+			fmt.Printf("%s\n", err.Error())
+			return
+		}
+		err = checkedDelete(recordsToDelete, cloudflareDNS)
+		if err != nil {
+			fmt.Printf("error deleting records: %s\n", err)
+		}
+	},
+}
+
+var dnsaddrTreeCreateCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Creates a tree of entries containing the multiaddrs at the provided root domain",
+	Long:  "Creates a tree of entries containing the multiaddrs at the provided root domain",
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(cmdMultiaddrs) == 0 {
+			fmt.Printf("must provide multiaddrs to put in the DNS records")
+			return
+		}
+		// Generate the dnsaddr entries required for the full tree
+		var dnsaddrsTo []string
+		for i := 0; i < len(cmdMultiaddrs)/nodeSize; i++ {
```

Comment:
I guess it's possible for the tree to be completely full, so no way to add another record without creating a new root?

Reply:
Not sure if you're talking about adding a single entry to an existing tree?

I didn't really see the value in implementing things like balancing or adding single entries at intermediate nodes. First because I don't see these lists having frequent churn, and second because this is just meant for testing so simple setup/teardown is sufficient.